### PR TITLE
Add uint16 output file type

### PR DESCRIPTION
**Pull request by Zcooger**
Because yes. Needed for LD Decode Player. https://github.com/happycube/ld-decode/issues/143

### CODE REVIEW

**Comment by Zcooger** [comment]
51691374_332537467356031_4210409586194120704_n

File: file.c [hunk 1]
```diff
@@ -73,6 +73,27 @@ static int _rf_file_write_int8_real(void *private, int16_t *iq_data, size_t samp
...
+{
+	rf_file_t *rf = private;
+	int16_t *u16 = rf->data;
+	int i;
+
+	while (samples)
+	{
+		for (i = 0; i < rf->samples && i < samples; i++, iq_data += 2)
+		{
+			u16[i] = (iq_data[0] + INT16_MAX);
```
Comment (fsphil) [inline file.c]: This should be `iq_data[0] - INT16_MIN`

File: file.c [hunk 2]
```diff
@@ -180,6 +201,15 @@ static int _rf_file_write_int8_complex(void *private, int16_t *iq_data, size_t s
 	return(HACKTV_OK);
 }
 
+static int _rf_file_write_uint16_complex(void *private, int16_t *iq_data, size_t samples)
+{
+	rf_file_t *rf = private;
+
+	fwrite(iq_data + INT16_MAX, sizeof(int16_t) * 2, samples, rf->f);
```
Comment (fsphil) [inline file.c]: This will write bad data, or crash. See `_rf_file_write_uint8_complex()` for an idea of how it should look.

Comment (Zcooger) [reply file.c]: Done too.

File: file.c (hunk 1 above)
Comment (Zcooger) [reply file.c]: Done.

**Comment by fsphil** [comment]
Tested OK and merged. Thanks!

**Pull request closed without merging**